core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -142,6 +149,9 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
         SqlSyntax.BINARY.unparse(writer, INTERSECT_DISTINCT, call, leftPrec, rightPrec);
       }
       break;
+    case REGEXP_SUBSTR:
```

Comment:
A Sqlkind is not needed for REGEXP_SUBSTR. The `unparse` should go `super.unparseCall(writer, call, leftPrec, rightPrec);`, in which it will require you to implement `unparse` for your new operators.